src/provider/sequencer.ts

```diff
@@ -592,7 +596,19 @@ export class SequencerProvider implements ProviderInterface {
   public async getStateUpdate(
     blockIdentifier: BlockIdentifier = this.blockIdentifier
   ): Promise<StateUpdateResponse> {
-    return this.fetchEndpoint('get_state_update', { blockIdentifier }).then(
+    const args: { blockHash?: string; blockNumber?: BlockNumber } = {};
```

Comment:
Add a utility getter to the `Block` class, something along the lines of:
```
get sequencerIdentifier() {
    return this.hash !== null 
      ? { blockHash: this.hash } 
      : { blockNumber: this.number ?? this.tag };
  }
```

So it can be used like `const args = new Block(blockIdentifier).sequencerIdentifier` (will probably need some type fiddling) and the `setIdentifier` mapping reused.